lib/resolv.rb

```diff
@@ -929,8 +929,11 @@ def initialize(host, port=Port)
         end
 
         def recv_reply(readable_socks)
-          len = readable_socks[0].read(2).unpack('n')[0]
+          len_data = readable_socks[0].read(2)
+          raise Errno::ECONNRESET if len_data.nil?
+          len = len_data.unpack('n')[0]
           reply = @socks[0].read(len)
+          raise Errno::ECONNRESET if reply.nil?
```

Comment:
While we're at it, could we also fix the cases where `len_data.bytesize != 2` and `reply.bytesize != len`?

Raising `Errno::ECONNRESET` when it doesn't involve the `ECONNRESET` errno feels wrong.